sessiontxn/isolation/readcommitted.go

```diff
@@ -104,6 +109,53 @@ func NeedSetRCCheckTSFlag(ctx sessionctx.Context, node ast.Node) bool {
 	return false
 }
 
+// NeedDisableWarmupInOptimizer checks whether optimizer calls `txnManger.AdviseWarmup()` to warmup in RC isolation.
+// txnManger.AdviseWarmup makes a tso request except that it's a readonly statement with RcReadCheckTS mode, please
+// reffer to tidb_rc_read_check_ts and know more about RcReadCheckTS mode. But for some special scenes, it maybe not
+// necessary to get tso from PD such as bellow.
+// 1. RC isolation && not internal sqls && In transaction
+// 2. Insert without select || point update/delete/lock-read
```

Comment:
Like RR isolation, TiDB doesn't get the latest ts from PD as for_update_ts but use the last ts of this transaction for point-lock-read, point-update, point-delete,  because they access only one row and decrease the conflict risk to very low.